### PR TITLE
feat(feature dev): Remove planning phase in favor for new code gen agent

### PR DESCRIPTION
**Pull request by willyyhuang**
## Problem

Us (QSDA) team is launching a new code generation agent.

We have conducted experiments wIth the new code generation agent, we observed:
- Removing planning phase produces higher quality code
- Removing planning phase has general better user experience and remove the inconsistency of code gen & plan gen

We have discussed with product that the rollout of new code generation agent should also include planning phase removal as a whole, without an A/B testing mechanism nor backward compatible implementation. The endpoint for plan generation will still exist to support older toolkit versions.

We plan to release the new code generation agent experience in the week of September 9th.

Please squash merge this pull request, thank you!

## Solution

- Removed planning phase & refine plan in a typical /dev flow.
- Modified customer facing copies.

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts [hunk 1]
```diff
@@ -172,23 +169,6 @@ export class FeatureDevController {
         const session = await this.sessionStorage.getSession(tabId)
 
         switch (session?.state.phase) {
```
Comment (chengoramazon) [inline packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:171]: We may no longer need the switch case here given that we added more points to allow users to vote feedback.

File: packages/core/src/amazonqFeatureDev/session/session.ts [hunk 2]
```diff
@@ -167,7 +145,6 @@ export class Session {
 
             // If approach was changed then we need to set it in the next state and this state
             this.state.approach = newApproach
```
Comment (chengoramazon) [inline packages/core/src/amazonqFeatureDev/session/session.ts:147]: Is the `state.approach` still being used?

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts (hunk 1 above)
Comment (willyyhuang) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:171]: okay

File: packages/core/src/amazonqFeatureDev/session/session.ts (hunk 2 above)
Comment (willyyhuang) [reply packages/core/src/amazonqFeatureDev/session/session.ts]: removed!

**Review by chengoramazon — APPROVED** [review]

**Comment by willyyhuang** [comment]
> Two questions:
> 
> 1. Are you going to update the e2e tests in https://github.com/aws/aws-toolkit-vscode/blob/master/packages/amazonq/test/e2e/amazonq/featureDev.test.ts (you could greatly reduce the number of test runs now that its just code generation)
> 2. Are all the UI text changes approved by a UX designer?

1. Ah, I can take a look at this, but could I follow up with another PR on this? As this is rather a large PR already, would be easier to review
2. Yes, all the text changes are approved by a tech writer, not UX designer

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts [hunk 3]
```diff
@@ -168,44 +165,23 @@ export class FeatureDevController {
         })
     }
 
-    private async processChatItemVotedMessage(tabId: string, messageId: string, vote: string) {
+    private async processChatItemVotedMessage(tabId: string, _: string, vote: string) {
```
[the remark this answers is not in the document]
Comment (willyyhuang) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:168]: sure

**Review by jpinkney-aws — APPROVED** [review]

**Pull request merged**